### ISSUE TITLE
 do you have some detail resource about quantize ?

### ISSUE THREAD

**Issue by victorygogogo** [pytorch/QNNPACK#3]
r[i] = scale * (q[i] – zero_point)

how to calculate the scale ,and do the scale need calibration ?

the input scale weight scale and the output scale ?? 
different input different scale  and the output have the different scale too ? 
how to solve this problem ? 

so do you have some detail resource about quantize ?

**Comment by Maratyszcza** [comment]
There are two ways to get quantized models for QNNPACK. You can start with a pre-trained floating-point models, and derive zero point and scale based on values in activation tensors (in the simplest implementation: based on min/max of activation tensors). Alternatively, you can simulate quantization during training by inserting fake quantization operators. The second approach typically results in much smaller accuracy loss.

I recommend this paper by Benoit Jacob et al as a reference: [
Quantization and Training of Neural Networks for Efficient Integer-Arithmetic-Only Inference](https://arxiv.org/abs/1712.05877).

**Comment by csuhawk** [comment]
@Maratyszcza  Will you going to open source QNNPACK quantize tool and methods? I'm looking forward for your open source.

**Comment by Maratyszcza** [comment]
@csuhawk We plan to release quantization tooling, but I can't share specific timeline when it happens.
